internal/access/credential.go

```diff
@@ -73,3 +79,71 @@ func UpdateCredential(c *gin.Context, user *models.Identity, newPassword string)
 
 	return nil
 }
+
+func lowercaseCheck(n int, password string) bool {
+	match, _ := regexp.MatchString(fmt.Sprintf(`(.*\p{Ll}){%d,}`, n), password)
```

Comment:
These regex rules are hard to read and maintain. It'll be easier to understand if it doesn't use raw regex for everything. Instead, it can use something like [regexp.FindAllStringIndex](https://pkg.go.dev/regexp#Regexp.FindAllStringIndex) on an expression like `[a-z]`. You can compare the output as `len(result) >= n`

Reply:
It would be great to avoid regex in general. I'm sure I could look up what all of this does, but it's generally not obvious form the regex string.

Could we use something like this instead?:

```go
var count int
for _, r := range password {
    if unicode.IsLower(r) {
        count++
    }
}
return count >= n
```

This pattern should work for uppercase and numbers as well (https://pkg.go.dev/unicode#IsNumber). https://pkg.go.dev/unicode#IsPunct or https://pkg.go.dev/unicode#IsSymbol may work for `symbolCheck`.

We can make it generic like this:

```go
func hasMinimumCount(min int, password string, check func(rune) bool) bool {
	var count int
	for _, r := range password {
		if check(r) {
			count++
		}
	}
	return count >= min
}
```

```go
if lowercase && !hasMinimumCount(minLowercase, password, unicode.IsLower) { ... }
```